src/azul/plugins/metadata/hca/transform.py

```diff
@@ -986,6 +989,14 @@ def _samples(self, samples: Iterable[api.Biomaterial]) -> MutableJSON:
                 assert False, sample
         return result
 
+    def _is_dcp_matrix(self, file: api.File) -> bool:
+        category = Submitter.category_for_file(file)
+        return category == SubmitterCategory.internal
```

Comment:
I think this method name is a bit misleading. A file with from an internal submitter aka source isn't automatically a matrix file. It used to be but we'll end up being wholly confused in the future unless we establish and equality between the Submitter enum and the file_source schema enum.

Also, we shouldn't hide our enum behind wrapper functions. The general approach should be to determine an enum instance once and them interrogate it.